packages/cli/src/push/abstract.push.ts

```diff
@@ -94,4 +94,17 @@ export abstract class AbstractPush<T> extends EventEmitter {
 
 		this.sendToSessions(type, data, userSessionIds);
 	}
+
+	/**
+	 * Closes all push existing connections
+	 */
+	closeAllConnections() {
+		for (const sessionId in this.connections) {
+			// Signal the connection that we want to close it.
+			// We are not removing the sessions here because it should be
+			// the implementation's responsibility to do so once the connection
+			// has actually closed.
+			this.close(this.connections[sessionId]);
+		}
```

Comment:
I don't know if we are buffering any push messages to need a proper `close` call. a very rude termination of websockets hasn't created any issues so far. Do we really need this?

Reply:
AFAIK if we don't close the websockets, the connections won't close until the HTTP server's timeout expires. That's because calling `.close()` on `net.server` only [stops accepting new connections](https://nodejs.org/docs/latest-v18.x/api/net.html#serverclosecallback). IMO it's always better to do graceful shutdown instead of hard termination